pkg/controller/v1alpha2/core/scopes/healthscope/healthscope.go

```diff
@@ -205,12 +232,51 @@ func CheckDaemonsetHealth(ctx context.Context, client client.Client, ref runtime
 		r.Diagnosis = errors.Wrap(err, errHealthCheck).Error()
 		return r
 	}
-	r.Target.UID = daemonset.GetUID()
+	compName, exist := daemonset.GetLabels()[oam.LabelAppComponent]
+	if !exist {
+		compName = "UNKNOWN NAME"
+	}
+	r.ComponentName = compName
+	r.TargetWorkload.UID = daemonset.GetUID()
+	r.Diagnosis = fmt.Sprintf(infoFmtReady, daemonset.Status.NumberReady, daemonset.Status.DesiredNumberScheduled)
 
+	// Health criteria
 	if daemonset.Status.NumberUnavailable != 0 {
-		r.Diagnosis = fmt.Sprintf(errFmtResourceNotReady, daemonset.Status)
 		return r
 	}
-	r.IsHealthy = true
+	r.HealthStatus = StatusHealthy
 	return r
 }
+
+// CheckByHealthCheckTrait checks health condition through HealthCheckTrait.
+func CheckByHealthCheckTrait(ctx context.Context, c client.Client, wlRef runtimev1alpha1.TypedReference, ns string) *HealthCondition {
+	//TODO(roywang) implement HealthCheckTrait feature
+	return nil
+}
+
+// CheckUnknownWorkload handles unknown type workloads.
+func CheckUnknownWorkload(ctx context.Context, c client.Client, wlRef runtimev1alpha1.TypedReference, ns string) *HealthCondition {
+	healthCondition := &HealthCondition{
+		TargetWorkload: wlRef,
+		HealthStatus:   StatusUnknown,
+		Diagnosis:      fmt.Sprintf(infoFmtUnknownWorkload, wlRef.APIVersion, wlRef.Kind),
+	}
+
+	wl := &unstructured.Unstructured{}
+	wl.SetGroupVersionKind(wlRef.GroupVersionKind())
+	if err := c.Get(ctx, client.ObjectKey{Namespace: ns, Name: wlRef.Name}, wl); err != nil {
+		healthCondition.Diagnosis += errors.Wrap(err, errHealthCheck).Error()
```

Comment:
We should not combine string,  I don't see we have more than one error message to combine?

Reply:
Yes, I will fix it.